services/QuillLMS/client/app/bundles/Teacher/components/classrooms/canvas_modal.tsx

```diff
@@ -0,0 +1,178 @@
+    header: ONE_STEP_AWAY_FROM_USING_CANVAS,
+    paragraphCopy: 'Looking to streamline your teaching process with Quill’s Canvas integration? Great news! Your School or District Premium subscription gives you access to the integration and it has been set up by you or another administrator for your school. The last step is for you to log in to Quill via your Canvas account. Once you have done so, you’ll be able\u00A0to:',
+    checklistItems: [
+      SEAMLESSLY_IMPORT_YOUR_CANVAS_ROSTERS,,
+      AUTOMATICALLY_CREATE_AND_SYNC_CANVAS_STUDENT_ACCOUNTS,
+    ],
+    buttons: [
+      <a className={PRIMARY_BUTTON_STYLE} href={HOW_TO_IMPORT_CANVAS_CLASSES_HREF} key={LEARN_HOW} rel="noopener noreferrer" target="_blank">{LEARN_HOW}</a>,
+    ]
+  }
+}
+
+interface CanvasModalProps {
+  close: () => void;
+  user: any;
+}
+
+const CanvasModal = ({ close, user }: CanvasModalProps) => {
+  const subscriptionType = user.subscription?.account_type
+  const hasSchoolOrDistrictPremium = subscriptionType?.includes(SCHOOL) || subscriptionType?.includes(DISTRICT)
+  const schoolLinkedToCanvas = user.school_linked_to_canvas
+  const userIsTeacher = user.role === TEACHER
+
+  let canvasState
+
+  if (userIsTeacher) {
+    if (hasSchoolOrDistrictPremium) {
+      // we don't show this modal if the user is already linked to Canvas and all of the other conditions are true, which would be state D
+      canvasState = schoolLinkedToCanvas ? TEACHER_LACKS_INDIVIDUAL_CANVAS : TEACHER_LACKS_SCHOOL_CANVAS;
+    } else {
+      canvasState = TEACHER_LACKS_RELEVANT_PREMIUM;
+    }
+  } else {
+    if (hasSchoolOrDistrictPremium) {
+      // we don't show this modal if the user is already linked to Canvas and all of the other conditions are true, which would be state H
+      canvasState = schoolLinkedToCanvas ? ADMIN_LACKS_INDIVIDUAL_CANVAS : ADMIN_LACKS_SCHOOL_CANVAS;
+    } else {
+      canvasState = ADMIN_LACKS_RELEVANT_PREMIUM;
+    }
+  }
+
+  return (
+    <div className="modal-container canvas-modal-container">
+      <div className="modal-background" />
+      <div className="canvas-modal quill-modal modal-body">
+        <div>
+          <h3 className="title">{canvasStates[canvasState].header}</h3>
+        </div>
+        <p>{canvasStates[canvasState].paragraphCopy}</p>
+        <ul>
+          {canvasStates[canvasState].checklistItems.map(checklistItem => (
+            <li key={checklistItem}>{circleCheckImg}{checklistItem}</li>
+          ))}
+        </ul>
+        {canvasStates[canvasState].asterisk}
+        <div className="form-buttons">
+          <button
+            className="quill-button outlined secondary medium focus-on-light"
+            onClick={close}
+            type="button"
+          >
+            Cancel
+          </button>
+          <div className="right-aligned-buttons">
+            {canvasStates[canvasState].buttons}
+          </div>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+export default CanvasModal;
```

Comment:
love the way how all of this is organized!

Reply:
Agreed.  Super easy to follow.